browser/android/youtube_script_injector/youtube_script_injector_tab_helper.cc

```diff
@@ -125,4 +120,46 @@ void YouTubeScriptInjectorTabHelper::PrimaryMainDocumentElementAvailable() {
   }
 }
 
+// static
+bool YouTubeScriptInjectorTabHelper::IsYouTubeVideo(const GURL& url) {
+  if (!url.is_valid() || url.is_empty()) {
+    return false;
+  }
+
+  // Check if domain is youtube.com (including subdomains).
+  if (!net::registry_controlled_domains::SameDomainOrHost(
+          url, GURL("https://www.youtube.com"),
+          net::registry_controlled_domains::INCLUDE_PRIVATE_REGISTRIES)) {
+    return false;
+  }
+
+  // Check if path is exactly "/watch" (case sensitive).
+  const auto path = url.path_piece();
+  constexpr std::string_view watch_path = "/watch";
+  if (path != watch_path) {
+    return false;
+  }
+
+  // Check if query exists and contains a non-empty "v" parameter.
+  const auto query = url.query_piece();
+  if (query.empty()) {
+    return false;
+  }
+
+  // Key-value pairs are '&' delimited and the keys/values are '=' delimited.
+  // Example: "https://www.youtube.com/watch?v=abcdefg&somethingElse=12345".
+  std::vector<std::pair<std::string_view, std::string_view>> key_value_pairs;
+  base::SplitStringIntoKeyValueViewPairs(query, '=', '&', &key_value_pairs);
```

Comment:
`url::ExtractQueryKeyValue`

Reply:
Replaced `SplitStringIntoKeyValueViewPairs` in favor of `ExtractQueryKeyValue` in commit [`daa76b0` (#29431)](https://github.com/brave/brave-core/pull/29431/commits/daa76b06ef964b5ed1f3f29b6eb809492fdcc5b2)